src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -166,29 +166,33 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
 
                 using (var process = Process.Start(processStartInfo))
                 {
-                    var finished = process.WaitForExit(timeOut);
-
-                    if (!finished)
+                    var errors = new StringBuilder();
+                    using (var errorTask = ConsumeStreamReaderAsync(process.StandardError, errors))
```

Comment:
In general I embrace deterministic disposal; however, after reading [this](https://blogs.msdn.microsoft.com/pfxteam/2012/03/25/do-i-need-to-dispose-of-tasks/), it seems generally not advantageous to dispose of tasks explicitly.  Task usage becomes more cumbersome.